processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationProcessorV2.java

```diff
@@ -89,25 +86,26 @@ public final void processNotifications(List<AlertNotificationModel> notification
                                                                                        .stream()
                                                                                        .map(filterableNotificationExtractor::wrapNotification)
                                                                                        .collect(Collectors.toList());
-        Map<DistributionJobModel, List<FilterableNotificationWrapper<?>>> jobsToNotifications = jobNotificationExtractor.mapJobsToNotifications(filterableNotifications, frequency);
-        for (Map.Entry<DistributionJobModel, List<FilterableNotificationWrapper<?>>> jobToNotifications : jobsToNotifications.entrySet()) {
-            processJobNotifications(jobToNotifications.getKey(), jobToNotifications.getValue());
+        Map<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper<?>>> jobsToNotifications = jobNotificationExtractor.mapJobsToNotifications(filterableNotifications, frequency);
+        for (FilteredDistributionJobResponseModel filteredDistributionJobResponseModel : jobsToNotifications.keySet()) {
```

Comment:
Would it be better to loop over the entry set, or leave it as is?

Reply:
Considering you call `jobsToNotifications.get(filteredDistributionJobResponseModel));` later on, it would be better to iterate over `.entrySet()`. That would be the most performant anyway while making the least assumptions about the Map implementation.